### DISCUSSION TITLE
Critical CSS

### DISCUSSION THREAD

**Discussion by CyrilKrylatov** [symfony/symfony#51664] [Q&A]
Hello everyone,

I'm trying to put together a performance strategy with [Critical CSS](https://web.dev/extract-critical-css/) with Symfony and Webpack Encore and make the whole thing working together without me.

So I'm trying to use [html-critical-webpack-plugin](https://github.com/anthonygore/html-critical-webpack-plugin) with the following configuration:

```js
const HtmlWebpackPlugin = require('html-webpack-plugin')
const HtmlCriticalWebpackPlugin = require('html-critical-webpack-plugin')
const path = require('path')
…
Encore
 .addPlugin(
    new HtmlWebpackPlugin({
      template: path.resolve(__dirname, 'public/homepage.html'),
      filename: path.resolve(__dirname, 'public/homepage_optimize.html')
    })
  )

  .addPlugin(() => {
    return new HtmlCriticalWebpackPlugin({
      base: path.resolve(__dirname, 'public'),
      src: 'homepage.html',
      dest: 'homepage_optimize.html',
      inline: true,
      minify: true,
      extract: true,
      width: 375,
      height: 565,
      penthouse: {
        blockJSRequests: false
      }
    })
  })
```

and my question here is: how can I make it works with an absolute URL and not a file? I want to provide him the URL of my homepage, my checkout page, my category page etc. and it manages to make it works?

Furthemore, for the people who achieved to put together a way to extract and load some Critical CSS within Symfony and Webpack Encore, how did you managed to make it work?

Thank you for your answers!

**Comment by sheeehy** [comment]
To handle absolute URLs instead of local files, you may need to modify your approach since html-critical-webpack-plugin primarily works with local HTML files. You can use a custom server-side solution or another npm package to download the HTML content from a URL, and then pass it to the plugin for processing.

You can use axios or node-fetch to fetch HTML from a URL, save it to a temporary location, and then process it with the plugins you mentioned.
